addons/mrp_subcontracting/wizard/mrp_product_produce.py

```diff
@@ -0,0 +1,85 @@
+
+    def _update_finished_move(self):
+        res = super(MrpProductProduce, self)._update_finished_move()
+        if not self.subcontract_move_id:
+            return res
+        if self.finished_lot_id:
+            move_lines = self.subcontract_move_id.move_line_ids.filtered(lambda ml: (ml.lot_id == self.finished_lot_id or not ml.lot_id) and
+            float_compare(ml.product_uom_qty, ml.qty_done, precision_rounding=ml.product_uom_id.rounding) > 0)
+        else:
+            move_lines = self.subcontract_move_id.move_line_ids.filtered(lambda ml: not ml.lot_id and
+            float_compare(ml.product_uom_qty, ml.qty_done, precision_rounding=ml.product_uom_id.rounding) > 0)
+
+        quantity = self.qty_producing
+        ml_to_unlink = self.env['stock.move.line']
+        for ml in move_lines:
+            rounding = ml.product_uom_id.rounding
+            if float_compare(quantity, 0, precision_rounding=rounding) <= 0:
+                break
+            quantity_to_process = min(quantity, ml.product_uom_qty - ml.qty_done)
+            quantity -= quantity_to_process
+
+            new_quantity_done = (ml.qty_done + quantity_to_process)
+
+            if float_compare(new_quantity_done, ml.product_uom_qty, precision_rounding=rounding) >= 0:
+                ml.write({
+                    'qty_done': new_quantity_done,
+                    'lot_id': self.finished_lot_id and self.finished_lot_id.id,
+                })
+            else:
+                new_qty_reserved = ml.product_uom_qty - new_quantity_done
+                default = {
+                    'product_uom_qty': new_quantity_done,
+                    'qty_done': new_quantity_done,
+                    'lot_id': self.finished_lot_id and self.finished_lot_id.id,
+                }
+                ml.copy(default=default)
+                if not float_is_zero(new_qty_reserved, precision_rounding=ml.product_uom_id.rounding):
+                    ml.with_context(bypass_reservation_update=True).write({
+                        'product_uom_qty': new_qty_reserved,
+                        'qty_done': 0
+                    })
+                else:
+                    ml_to_unlink |= ml
+        if ml_to_unlink:
+            ml_to_unlink.unlink()
+
+        if float_compare(quantity, 0, precision_rounding=self.subcontract_move_id.product_uom.rounding) > 0:
+            self.env['stock.move.line'].create({
+                'move_id': self.subcontract_move_id.id,
+                'picking_id': self.subcontract_move_id.picking_id.id,
+                'product_id': self.product_id.id,
+                'location_id': self.subcontract_move_id.location_id.id,
+                'location_dest_id': self.subcontract_move_id.location_dest_id.id,
+                'product_uom_qty': 0,
+                'product_uom_id': self.product_uom_id.id,
+                'qty_done': quantity,
+                'lot_id': self.finished_lot_id and self.finished_lot_id.id,
```

Comment:
maybe just unlink all the reserved move lines instead of trying to synchronize it?